torchvision/transforms/functional.py

```diff
@@ -285,11 +285,10 @@ def to_pil_image(pic, mode=None):
         if pic.shape[-1] > 4:
             raise ValueError(f"pic should not have > 4 channels. Got {pic.shape[-1]} channels.")
 
-    npimg = pic
-    if isinstance(pic, torch.Tensor):
-        if pic.is_floating_point() and mode != "F":
-            pic = pic.mul(255).byte()
-        npimg = np.transpose(pic.cpu().numpy(), (1, 2, 0))
+    npimg = np.transpose(pic.cpu().numpy(), (1, 2, 0)) if isinstance(pic, torch.Tensor) else pic
+
+    if np.issubdtype(npimg.dtype, np.floating) and mode != "F":
```

Comment:
There's not such thing as `is_floating_types` for numpy arrays as far as I know, so I went for the above, found in https://github.com/numpy/numpy/issues/17325